src/Enum.php

```diff
@@ -27,6 +27,16 @@ abstract class Enum implements JsonSerializable
     /** @psalm-var array<string, array<string, \Spatie\Enum\EnumDefinition>>  */
     private static array $definitionCache = [];
 
+    /**
+     * @return static[]
+     */
+    public static function all(): array
+    {
+        $instances = array_map(fn (EnumDefinition $definition) => static::make($definition->value), static::resolveDefinition());
```

Comment:
```suggestion
        $instances = array_map(
            fn (EnumDefinition $definition): self => static::make($definition->value), 
            static::resolveDefinition()
        );
```

Reply:
do you want to return `self` when it in fact returns `static`? Maybe instead it should be `: Enum` instead to match the return type of `static::make()`?

Reply:
Good point. `static` is a PHP8 return type, but if we've used `Enum` in the other methods we should stick with it! 👌